cfme/tests/automate/test_simulation.py

```diff
@@ -231,3 +231,58 @@ def test_attribute_value_message(custom_instance):
             request="call_instance_with_message",
             execute_methods=True,
         )
+
+
+@pytest.mark.meta(automates=[1672007])
+def test_action_invoke_custom_automation(request, appliance):
+    """
+    Bugzilla:
+        1672007
+
+    Polarion:
+        assignee: ghubale
+        initialEstimate: 1/8h
+        caseposneg: positive
+        casecomponent: Automate
+        testSteps:
+            1. Navigate to Control > explorer > actions
+            2. Select 'add a new action' from configuration dropdown
+            3. Add description and select 'Action Type' - Invoke custom automation
+            4. Fill attribute value pairs and click on add
+            5. Edit the created action and add new attribute value pair
+            6. Remove that newly added attribute value pair before clicking on save and then click
+               on save
+        expectedResults:
+            1.
+            2.
+            3.
+            4.
+            5. Save button should enabled
+            6. Action should be saved successfully
+    """
+    attr_val = [{
+        "attribute_1": fauxfactory.gen_alpha(),
+        "attribute_2": fauxfactory.gen_alpha(),
+        "attribute_3": fauxfactory.gen_alpha(),
+        "attribute_4": fauxfactory.gen_alpha(),
+        "attribute_5": fauxfactory.gen_alpha(),
+    } for _ in range(2)]
+
+    automation_action = appliance.collections.actions.create(
+        fauxfactory.gen_alphanumeric(),
+        "Invoke a Custom Automation",
+        dict(
+            message=fauxfactory.gen_alpha(),
+            request=fauxfactory.gen_alpha(),
+            avp=attr_val[0]
+        )
+    )
+    request.addfinalizer(automation_action.delete_if_exists)
+
+    view = navigate_to(automation_action, "Edit")
+    view.avp.fill(attr_val[1])
+    assert view.save_button.is_enabled
+    view.avp.clear()
+    view.save_button.click()
+    view = navigate_to(automation_action, "Details")
```

Comment:
I am not getting at @valaparthvi 
- I used it because of opinion i got from others like it would be better to navigate to that view rather than creating the view. So navigating to that view confirms **is_displayed**. So keeping as it is.

Reply:
The reason I suggested `create_view` is, in case the Details page is not displayed on clicking the save button, you will be navigated to that page, and any flash on the page displayed after you click on the save button will disappear. So creating a view will ensure that any flash message displayed does not disappear, regardless of what `view.is_displayed` is.

Reply:
I Agree! This is what I was also thinking and creating view. It will be better to have more opinions @digitronik @mshriver on this.

Reply:
yup, view creation is better option.
we use navigation when we want that nav to specific view. but here you are expecting that view after form submit and like to assert flash.
I would suggest to put `wait` as well so it will add one more check in test means after submit it should land on details page. I faced some bug where add/edit like pages land on All page after submission.

Reply:
Hi @digitronik 
I tried with ```wait``` in create_view but after editing action and saving with updated values ends up failure in ```is_displayed``` with below error:
```

self = <cfme.control.explorer.actions.ActionDetailsView object at 0x7fd271e59150>

    @property
    def is_displayed(self):
        import ipdb;ipdb.set_trace()
        return (
            self.in_control_explorer and
>           self.title.text == 'Action "{}"'.format(self.context["object"].description) and
            self.actions.is_opened and
            self.actions.tree.currently_selected == [
                "All Actions",
                self.context["object"].description
            ]
        )
E       KeyError: 'object'

../../control/explorer/actions.py:126: KeyError

````
- So I have removed wait from create_view and it works as expected.

Reply:
I would use `obj.create_view`